File: berkeley/ruby/update/berkeley_update.rb
#!/usr/bin/ruby1.8
# -*- coding: utf-8 -*-
#
#	berkeley_update.rb
#
#					Mar/21/2013
#    
require 'bdb'
require	'json'
#
load '/var/www/data_base/common/ruby_common/kvalue_manipulate.rb'
# --------------------------------------------------------------------
puts "*** 開始 ***"
db_file = ARGV[0]
id_in = ARGV[1]
population_in = ARGV[2].to_i

puts id_in,population_in
#
puts "*** ppp ***"
#ee = BDB::Btree::open(db_file,"table1","w",0777)
ee = BDB::Btree::open(db_file,"table1","r+",0777)
puts "*** qqq ***"
#
kvalue_update_proc(ee,id_in,population_in)
#
ww=ee.to_hash
kvalue_display_proc(ww)
ee.close()
#

print "ww.size =",ww.size,"\n"
puts "*** 終了 ***"

File: berkeley/python/update/berkeley_update.py
#! /usr/bin/python3
# -*- coding: utf-8 -*-
#
#	berkeley_update.py
#
#					Jun/11/2015
import	sys
import json
from bsddb3 import db
import string
#
sys.path.append ('/var/www/data_base/common/python_common')
from dbm_manipulate import dbm_update_proc
# -------------------------------------------------------------
print ("*** 開始 ***")
#
db_name = sys.argv[1]
key_in = sys.argv[2]
population_in = int (sys.argv[3])
print ("%s\t%d" % (key_in, population_in))
#
adb = db.DB ()
adb.open (db_name)
dbm_update_proc	(adb,key_in,population_in)
#
adb.close ()
#
print	("*** 終了 ***")
# -------------------------------------------------------------
